pyaedt/modules/Boundary.py

```diff
@@ -356,47 +356,191 @@ def delete(self):
+                    solderbumps_modeling, ", ".join(list(self._solderbumps_map.keys()))
+                )
+            )
+            return False
+
+        update_properties = {
+            "CreateTopSolderballs": modelling is not None,
+            "TopConnectorType": modelling,
+            "TopSolderballsModelType": self._solderbumps_map[solderbumps_modeling],
+            "BondwireMaterial": bondwire_material,
+            "BondwireDiameter": bondwire_diameter,
+        }
+
+        self.pcb.props["NativeComponentDefinitionProvider"].update(update_properties)
+        return True
+
+    def __repr__(self):
+        return "Package"
+
+
+class PCBSettingsDeviceParts(object):
+    def __init__(self, pcb_obj, app):
+        self._app = app
+        self.pcb = pcb_obj
         self._filter_map2name = {"Cap": "Capacitors", "Ind": "Inductors", "Res": "Resistors"}
 
+    def __eq__(self, other):
+        if isinstance(other, str):
+            return other == "Device"
+        elif isinstance(other, self.__class__):
+            return self.__dict__ == other.__dict_
+        else:
+            return False
+
+    def __ne__(self, other):
+        return not self.__eq__(other)
+
+    def __repr__(self):
+        return "Device"
+
+    @property
+    @pyaedt_function_handler()
+    def simplify_parts(self):
+        """Get whether parts are simplified  as cuboid."""
+        return self.pcb.props["NativeComponentDefinitionProvider"]["ModelDeviceAsRect"]
+
+    @simplify_parts.setter
+    @pyaedt_function_handler()
+    def simplify_parts(self, value):
+        """Set whether parts are simplified as cuboid.
+
+        Parameters
+        ----------
+        value : bool
+            Whether parts are simplified as cuboid.
+        """
+        self.pcb.props["NativeComponentDefinitionProvider"]["ModelDeviceAsRect"] = value
+
+    @property
+    @pyaedt_function_handler()
+    def surface_material(self):
+        """Surface material to apply to parts."""
+        return self.pcb.props["NativeComponentDefinitionProvider"]["DeviceSurfaceMaterial"]
+
+    @surface_material.setter
+    @pyaedt_function_handler()
+    def surface_material(self, value):
+        """Set surface material to apply to parts.
+
+        Parameters
+        ----------
+        value : str
+            Surface material to apply to parts.
+        """
+        self.pcb.props["NativeComponentDefinitionProvider"]["DeviceSurfaceMaterial"] = value
+
     @property
     @pyaedt_function_handler()
     def footprint_filter(self):
         """Minimum component footprint for filtering."""
-        if self.props["NativeComponentDefinitionProvider"]["PartsChoice"] != 1:
+        if self.pcb.props["NativeComponentDefinitionProvider"]["PartsChoice"] != 1:
             self._app.logger.error(
                 "Device Parts modeling is not active, hence no filtering or override option is available."
```

Comment:
```suggestion
                "Device parts modeling is not active.  No filtering or override option is available."
```
The message used unnecessary capitalization and was written as a run-on sentence.